examples/feature_selection/plot_f_test_vs_mi.py

```diff
@@ -0,0 +1,47 @@
+"""
+===========================================
+Comparison of F-test and mutual information
+===========================================
+
+This example illustrates the differences between univariate F-test statistics
+and mutual information.
+
+We consider 3 features x_1, x_2, x_3 distributed uniformly over [0, 1], the
+target depends on them as follows:
+
+y = x_1 + sin(6 * pi * x_2) + 0.1 * N(0, 1), that is the third features is completely irrelevant.
+
+The code below plots the dependency of y against individual x_i and normalized
+values of univariate F-tests statistics and mutual information.
+
+As F-test captures only linear dependency, it rates x_1 as the most
```

Comment:
sounds great, thank you very much, but is about over fitting example, do
you have one?

On Fri, Jan 15, 2016 at 10:48 AM, Nikolay Mayorov <notifications@github.com>
wrote:

> In examples/feature_selection/plot_f_test_vs_mi.py
> <https://github.com/scikit-learn/scikit-learn/pull/5372#discussion_r49867717>
> :
>
> > +===========================================
> > +Comparison of F-test and mutual information
> > +===========================================
> > +
> > +This example illustrates the differences between univariate F-test statistics
> > +and mutual information.
> > +
> > +We consider 3 features x_1, x_2, x_3 distributed uniformly over [0, 1], the
> > +target depends on them as follows:
> > +
> > +y = x_1 + sin(6 * pi * x_2) + 0.1 * N(0, 1), that is the third features is completely irrelevant.
> > +
> > +The code below plots the dependency of y against individual x_i and normalized
> > +values of univariate F-tests statistics and mutual information.
> > +
> > +As F-test captures only linear dependency, it rates x_1 as the most
>
> Honestly, I think someone needs to write a section explaining what are
> F-tests, chi2 (and mutual info), whey there applicable and how they
> different. But I suggest to delegate it to another PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/5372/files#r49867717>.
>